src/engine/nodes/BaseProcessor.h

```diff
@@ -62,6 +63,7 @@
 #define EL_INTERNAL_UID_MIDI_MONITOR             1016
 #define EL_INTERNAL_UID_OSC_RECEIVER             1017
 #define EL_INTERNAL_UID_OSC_SENDER               1018
+#define EL_INTERNAL_UID_EQ_FILTER                1017
```

Comment:
@mfisher31 Looks like here's a merge conflict: the UIDs of OSCReceiver and EQFilter are the same 1017.

Reply:
Nice work on those OSC nodes man!  There's definitely some things that'll need tweaked for realtime consideration.  No biggie though, this is an awesome start.  Always have to remember that things like `std::vector<T>::push_back(...)` dynamically allocate memory.  

Same thing with string building... these are killers in the RT thread.  No problem though,  I think we can use a semaphore + alternate thread to do the OSC/MIDI processing in.  Using the semaphore literally will make these introduce near-zero overhead.

Reply:
and by the way, not bashing about the RT thread stuff.  This is awesome work, just need to make sure absolutely no dynamic allocations are happening, that's all.

Reply:
Thank you for the feedback. Yes, I'm sure there are places that need to be smoothed over with careful eyes. In particular, I'm not confident that there aren't any memory leaks or thread access issues.

As you can see from the code, my approach is quite primitive, since I'm still a beginner with C++. I figured the "grunt work" would help anyway, like mapping OSC <-> MIDI messages. I'm hoping you can find more efficient/elegant and safe ways of doings things.

Reply:
Sall good!  Didn't see anything that'd cause a memory leak.  I remember years ago I asked the [LV2] mailing list why I shouldn't use std::vector in a realtime thread... drobilla kindly slapped me on the wrist... hehehe  "because it allocates!"

Reply:
I've picked up a few textbooks on C++ for bedtime reading, hopefully I'll have a better grasp of how memory/thread management should work.

I was reading the documentation for `boost/signal2`. Since it's already included in Element, It seemed like a good way to (possibly) simplify inter-thread communication. I'm studying JUCE's `Thread` class too, I wanted to use it for OSC message processing - but it was still a bit beyond my skills.

Reply:
Yep, boost signals have an internal mutex so they are internally thread safe.  But say thread A triggers it and thread B is listening, the function call still comes from thread A.  Stuff can get tricky!